src/Runtime/ExecutionSession.cpp

```diff
@@ -135,26 +112,21 @@ OMTensorList *ExecutionSession::run(OMTensorList *input) {
            << errMessageStr << "'" << std::endl;
     throw std::runtime_error(errStr.str());
   }
+  errno = 0; // No errors.
   return output;
 }
 
 const std::string ExecutionSession::inputSignature() const {
-  if (!_entryPointFunc) {
-    std::stringstream errStr;
-    errStr << "Must set the entry point before calling signature function"
-           << std::endl;
-    throw std::runtime_error(errStr.str());
-  }
+  if (!_entryPointFunc)
+    throw std::runtime_error(reportMissingEntryPointError("signature"));
```

Comment:
Is "signature" the actual name of the entry point ?

Reply:
No, before running certain commands, the user must define which entry point this is for. If the default is `run_main_graph`, then signatures and run go towards that one. If there are multiple entry points, the user must first indicate which of the many entry point to use BEFORE calling the run/signature calls.